native/acter/src/api/client/sliding_sync.rs

```diff
@@ -452,22 +455,35 @@ impl Client {
                 );
 
                 if inner.is_space() {
-                    if let Some(space_idx) = spaces.iter().position(|s| s.room_id() == r_id) {
+                    let pos = {
+                        let mut spaces = self.spaces.read().await;
+                        spaces.iter().position(|s| s.room_id() == r_id)
+                    };
+                    if let Some(space_idx) = pos {
+                        let mut spaces = self.spaces.write().await;
```

Comment:
this literally means that between the first read and the second write the position could have been changed. the point of the mutexes is, however, to ensure consistency. which this doesn't.

Reply:
Fixed by https://github.com/acterglobal/a3/pull/2522/commits/efc5dc447eeba2a4904c29735468c1169816caed